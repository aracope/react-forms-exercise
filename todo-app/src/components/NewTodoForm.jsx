import React from "react";
import { useFormik } from "formik";

function NewTodoForm({ addTodo }) {
  const formik = useFormik({
    initialValues: { task: "" },
    onSubmit: (values, { resetForm }) => {
      addTodo(values.task);
      resetForm();
    },
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <label htmlFor="task">Task:</label>
      <input
        id="task"
        name="task"
        type="text"
        placeholder="Enter a task"
        aria-label="Task"
        onChange={formik.handleChange}
        value={formik.values.task}
      />
      <button type="submit">Add</button>
    </form>
  );
}

export default NewTodoForm;