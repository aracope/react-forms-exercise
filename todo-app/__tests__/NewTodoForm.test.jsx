import React from "react";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NewTodoForm from "../src/components/NewTodoForm";

it("renders without crashing", () => {
  render(<NewTodoForm addTodo={() => { }} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => { }} />);
  expect(asFragment()).toMatchSnapshot();
});

it("submits form and calls addTodo", async () => {
  const mockAdd = jest.fn();
  render(<NewTodoForm addTodo={mockAdd} />);
  
  await userEvent.type(screen.getByLabelText(/task/i), "Walk dog"); 
  await userEvent.click(screen.getByText("Add"));                    

  expect(mockAdd).toHaveBeenCalledWith("Walk dog");
});
