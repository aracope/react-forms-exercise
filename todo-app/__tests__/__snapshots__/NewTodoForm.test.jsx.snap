// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task:
    </label>
    <input
      aria-label="Task"
      id="task"
      name="task"
      placeholder="Enter a task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
